src/components/FullscreenImage/FullscreenImage.tsx

```diff
@@ -15,19 +18,38 @@ export interface FullscreenImageProps extends ImageProps {
     modalImageClass?: string;
     imageStyle?: CSSProperties;
     extraProps?: HTMLProps<HTMLDivElement>;
+    sliderData?: {items: ModelImageProps[]; initialIndex: number};
 }
 
 const b = block('fullscreen-image');
 const FULL_SCREEN_ICON_SIZE = 18;
 const CLOSE_ICON_SIZE = 24;
 
 const FullscreenImage = (props: FullscreenImageProps) => {
-    const {imageClassName, modalImageClass, imageStyle, alt = i18n('img-alt'), extraProps} = props;
+    const {
+        imageClassName,
+        sliderData,
+        modalImageClass,
+        imageStyle,
+        alt = i18n('img-alt'),
+        extraProps,
+    } = props;
     const [isOpened, setIsOpened] = useState(false);
+    const [sliderLoaded, setSliderLoaded] = useState(false);
 
     const openModal = () => setIsOpened(true);
     const closeModal = () => setIsOpened(false);
 
+    useEffect(() => {
+        if (!isOpened) {
```

Comment:
add condition if we have sliderData, because we dont need trigger this hook without slider